src/coreclr/tools/Common/TypeSystem/Interop/IL/MarshalUtils.cs

```diff
@@ -23,7 +23,10 @@ public static bool IsBlittableType(TypeDesc type)
                 && !baseType.IsWellKnownType(WellKnownType.Object)
                 && !baseType.IsWellKnownType(WellKnownType.ValueType);
 
-            if (hasNonTrivialParent && !IsBlittableType(baseType))
+            // Type is blittable only if parent is also blittable and is not empty.
+            if (hasNonTrivialParent && !IsBlittableType(baseType) ||
+                (hasNonTrivialParent && baseType is MetadataType baseMdType &&
```

Comment:
Nit: `baseType` can be declared as `DefType` and then the cast to `MetadataType` will be unnecessary because the byte counts are available on `DefType`.

Reply:
Changed to `DefType`.

Reply:
Oh, I meant to change:

```
TypeDesc baseType = type.BaseType;
```

(above, so can't comment on that directly) into:

```
DefType baseType = type.BaseType;
```

Reply:
Cool. Updated and simplified the condition. There might be a better way to get the size without padding, but I tried several options and arithmetic operation seemed to be necessary to get LayoutInt.Zero for empty class having sequential layout. Approach also works for explicit layout.